test/jest/__tests__/utils/utils.DependencyUtils.ts.spec.ts

```diff
@@ -0,0 +1,597 @@
+                versions: [
+                    {version: "1.0.0", dependencies: ["author-modWithLoopDependencyB-1.0.0"]},
+                    {version: "2.0.0", dependencies: ["author-modWithLoopDependencyB-1.0.0"]}
+                ]
+            });
+            const modWithLoopDependencyB = addMockPackage({
+                name: "modWithLoopDependencyB",
+                dependencies: ["author-modWithLoopDependencyA-2.0.0"],
+                versions: [
+                    {version: "2.0.0", dependencies: ["author-modWithLoopDependencyA-2.0.0"]},
+                    {version: "1.0.0", dependencies: ["author-modWithLoopDependencyA-2.0.0"]}
+                ]
+            });
+
+            const combos = [modWithLoopDependencyA];
+            expectToContainOnly(await getFullDependencyList(combos, game, [], installMode), expected);
+        });
+    });
+
+
+    describe("A mod with a loop dependency, starting from the mod with a dependency to newer version", () => {
+        it.each([
+            [InstallMode.INSTALL_SPECIFIC, ["author-modWithLoopDependencyB-1.0.0", "author-modWithLoopDependencyA-2.0.0"]],
+            [InstallMode.UPDATE_ALL, ["author-modWithLoopDependencyB-1.0.0", "author-modWithLoopDependencyA-2.0.0"]],
+        ])
+        ("", async (installMode, expected) => {
+            const modWithLoopDependencyA = addMockPackage({
+                name: "modWithLoopDependencyA",
+                dependencies: ["author-modWithLoopDependencyB-1.0.0"],
+                versions: [
+                    {version: "2.0.0", dependencies: ["author-modWithLoopDependencyB-1.0.0"]},
+                    {version: "1.0.0", dependencies: ["author-modWithLoopDependencyB-1.0.0"]}
+                ]
+            });
+            const modWithLoopDependencyB = addMockPackage({
+                name: "modWithLoopDependencyB",
+                dependencies: ["author-modWithLoopDependencyA-2.0.0"],
+                versions: [
+                    {version: "1.0.0", dependencies: ["author-modWithLoopDependencyA-2.0.0"]},
+                    {version: "2.0.0", dependencies: ["author-modWithLoopDependencyA-2.0.0"]}
+                ]
+            });
+
+            const combos = [modWithLoopDependencyB];
+            expectToContainOnly(await getFullDependencyList(combos, game, [], installMode), expected);
+        });
+    });
+
+
+    describe("A mod with a loop dependency, but one of the dependencies is a mod pack, starting from the mod pack", () => {
+        it.each([
+            [InstallMode.INSTALL_SPECIFIC, ["author-modWithLoopDependencyButOneOfThemIsAModPackA-1.0.0", "author-modWithLoopDependencyButOneOfThemIsAModPackB-1.0.0"]],
+            [InstallMode.UPDATE_ALL, ["author-modWithLoopDependencyButOneOfThemIsAModPackA-1.0.0", "author-modWithLoopDependencyButOneOfThemIsAModPackB-2.0.0"]],
+        ])
+        ("", async (installMode, expected) => {
+            const modWithLoopDependencyButOneOfThemIsAModPackA = addMockPackage({
```

Comment:
Nitpick: `modWithLoopDependency` and `modpackWithLoopDependency` would be clearer names here and in the next test.